File: week-1/sentimental-mario-less/mario.py
def get_int(prompt):
  while True:
    try:
      value = int(input(prompt))
      if 0 < value <= 8:
        return value
      else:
        print("Please enter a positive integer between 1 and 8.")
        print("Try again")
    except ValueError:
      print("Invalid input. Please enter a valid integer.")

def generate_pyramid(height):
  for i in range(1, height + 1):
    spaces = height - i
    blocks = i

    for _ in range(spaces):
      print(" ", end="")

    for _ in range(blocks):
      print("#", end="")

    print()

def main():
  height = get_int("Enter the height of the half-pyramid (from 1 to 8): ")
  generate_pyramid(height)

if __name__ == "__main__":
  main()

File: week-1/sentimental-mario-less/mario-less/mario.c
//#include <cs50.h>
//#include <stdio.h>
//
//int main(void)
//{
//    int height;
//    do
//    {
//        height = get_int("Height: ");
//    }
//    while (1 > height || height > 8);
//
//    for (int i = 0; i < height; i++)
//    {
//        for (int j = 0; j < height - i - 1; j++)
//        {
//            printf(" ");
//        }
//
//        for (int k = 0; k < i + 1; k++)
//        {
//            printf("#");
//        }
//
//        printf("\n");
//    }
//
//    return 0;
//}


#include <cs50.h>
#include <stdio.h>

void draw(int n);

int main(void)
{
    int hight = get_int("Enter the height of your pyramid: ");
    draw(hight);
}

void draw(int n)  //5
{
    for (int i = 0; i < n; i++)
    {// i=1
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}
